### DISCUSSION TITLE
asynchronous custom validation rules

### DISCUSSION THREAD

**Discussion by erwinstone** [jadKhoury1/simple-body-validator#25] [Ideas]
Is there a plan to add asynchronous custom validation rules?
It will be very useful, especially for validation related to checking the database

**Comment by jadKhoury1** [comment]
Hello, yes you are right, this is an important feature and it is actually the next feature on the list. However, we cannot commit to a specific timeline for now, because the work related to the library is being done on free time.

**Comment by jadKhoury1** [comment]
Hello you can now register custom rules that return a **Promise**, when you do so you can use the **validateAsync** method instead of the **validate** method to run your validations. You can read more about it here https://www.simple-body-validator.com/asynchronous-validation

**Comment by erwinstone** [comment]
Hello,
Thanks for adding it. It's great to see it in action now
